### PR TITLE
Dev/indirect blocks

### PR DESCRIPTION
**Pull request by PigBrainOverflow**
Bug fixed. Still needs more testing.

### CODE REVIEW

File: src/uwufs/cpp/INode.cpp [hunk 1]
```diff
@@ -126,6 +128,7 @@ uwufs_blk_t INode::remove_dblk(uwufs_inode *inode, int device_fd, uwufs_blk_t in
         return 0;
     }
     if (index < LEVEL_0_BLOCKS) {   // direct block
+		// XXX: erase entry here too?
```
Comment (yuheho7749) [inline src/uwufs/cpp/INode.cpp:131]: ^

File: src/uwufs/cpp/INode.cpp [hunk 2]
```diff
@@ -113,6 +114,7 @@ std::pair<uwufs_blk_t, bool> INode::recursive_remove_dblk(int device_fd, uint8_t
 #endif
         return {block_no, true};
     }
+	// XXX: erase entry here?
```
Comment (yuheho7749) [inline src/uwufs/cpp/INode.cpp:117]: ^

File: src/uwufs/cpp/INode.cpp [hunk 3]
```diff
@@ -97,6 +97,7 @@ std::pair<uwufs_blk_t, bool> INode::recursive_remove_dblk(int device_fd, uint8_t
 #endif
             return {block_no, true};
         }
+		// XXX: erase entry here?
```
Comment (yuheho7749) [inline src/uwufs/cpp/INode.cpp:100]: ^

**Comment by yuheho7749** [comment]
This PR is massive already to change everything to use indirect blocks. Most things are fixed but there is still one (known) bug #33 to be fixed later. The tempfix is to immediately return/not use append_dblk if there are less than 5-7ish free data blks.

**Review by kkrachenfels — APPROVED** [review]

**Pull request merged**